plugin/trino-base-jdbc/src/main/java/io/trino/plugin/jdbc/BaseJdbcClient.java

```diff
@@ -551,6 +552,22 @@ protected static Optional<ParameterizedExpression> getAdditionalPredicate(List<P
                         .collect(toImmutableList())));
     }
 
+    protected JdbcTypeHandle getJdbcTypeHandle(ConnectorSession session, JdbcTableHandle tableHandle, ParameterizedExpression parameterizedExpression)
```

Comment:
I'm not sure if it is a efficient way to fetch the types of a synthetic column - we could also try to batch them if this approach is fine. cc: @findepi

Reply:
as discussed offline. this method will not exist in the final implementation